packages/ui-components/src/GooglePay/index.tsx

```diff
@@ -40,7 +41,13 @@ export function GooglePay({ config }: GooglePayProps) {
     async function onLoad() {
       const appConfig = await getAppSDKConfig(app);
       const paymentsClient = new google.payments.api.PaymentsClient({
-        environment: appConfig.isSandbox ? "TEST" : "PRODUCTION",
+        // Always use 'test' in staging, but use the resolved environment in production
+        environment:
+          apiConfig.environment === "staging"
+            ? "TEST"
+            : appConfig.isSandbox
+            ? "TEST"
+            : "PRODUCTION",
```

Comment:
```suggestion
environment: (appConfig.isSandbox || !apiConfig.environment === 'production') ? "TEST" : "PRODUCTION"
```

I'd be inclined to avoid the nested ternary, but that's purely preference, non-blocking

Reply:
I think while it's less code, I'd argue it hides the logic we want to represent.  I think I'll leave it as-is for now.